packages/contracts-bedrock/tasks/check-l2.ts

```diff
@@ -0,0 +1,628 @@
+    const SequencerFeeVault = await hre.ethers.getContractAt(
+      'SequencerFeeVault',
+      predeploys.SequencerFeeVault
+    )
+
+    const version = await SequencerFeeVault.version()
+    assertSemver(version, 'SequencerFeeVault')
+
+    const RECIPIENT = await SequencerFeeVault.RECIPIENT()
+    if (RECIPIENT === hre.ethers.constants.AddressZero) {
+      throw new Error('undefined RECIPIENT')
+    }
+    logLoud()
+    console.log(`  - RECIPIENT: ${RECIPIENT}`)
+    logLoud()
+
+    const l1FeeWallet = await SequencerFeeVault.l1FeeWallet()
+    if (l1FeeWallet === hre.ethers.constants.AddressZero) {
+      throw new Error('undefined l1FeeWallet')
+    }
+    console.log(`  - l1FeeWallet: ${l1FeeWallet}`)
+
+    const MIN_WITHDRAWAL_AMOUNT =
+      await SequencerFeeVault.MIN_WITHDRAWAL_AMOUNT()
+    console.log(`  - MIN_WITHDRAWAL_AMOUNT: ${MIN_WITHDRAWAL_AMOUNT}`)
+
+    await checkProxy(hre, 'SequencerFeeVault')
+    await assertProxy(hre, 'SequencerFeeVault')
+  },
+  // OptimismMintableERC20Factory
+  // - check version
+  OptimismMintableERC20Factory: async (hre: HardhatRuntimeEnvironment) => {
+    const OptimismMintableERC20Factory = await hre.ethers.getContractAt(
+      'OptimismMintableERC20Factory',
+      predeploys.OptimismMintableERC20Factory
+    )
+
+    const version = await OptimismMintableERC20Factory.version()
+    assertSemver(version, 'OptimismMintableERC20Factory', '1.0.0')
+
+    const BRIDGE = await OptimismMintableERC20Factory.BRIDGE()
+    if (BRIDGE === hre.ethers.constants.AddressZero) {
+      throw new Error('BRIDGE misconfigured')
+    }
+
+    await checkProxy(hre, 'OptimismMintableERC20Factory')
+    await assertProxy(hre, 'OptimismMintableERC20Factory')
+  },
+  // L1BlockNumber
+  // - check version
+  L1BlockNumber: async (hre: HardhatRuntimeEnvironment) => {
+    const L1BlockNumber = await hre.ethers.getContractAt(
+      'L1BlockNumber',
+      predeploys.L1BlockNumber
+    )
+
+    const version = await L1BlockNumber.version()
+    assertSemver(version, 'L1BlockNumber')
+
+    await checkProxy(hre, 'L1BlockNumber')
+    await assertProxy(hre, 'L1BlockNumber')
+  },
+  // L1Block
+  // - check version
+  L1Block: async (hre: HardhatRuntimeEnvironment) => {
+    const L1Block = await hre.ethers.getContractAt(
+      'L1Block',
+      predeploys.L1Block
+    )
+
+    const version = await L1Block.version()
+    assertSemver(version, 'L1Block')
+
+    await checkProxy(hre, 'L1Block')
+    await assertProxy(hre, 'L1Block')
+  },
+  // LegacyERC20ETH
+  // - not behind a proxy
+  // - check name
+  // - check symbol
+  // - check decimals
+  // - check BRIDGE
+  // - check REMOTE_TOKEN
+  // - totalSupply should be set to 0
+  LegacyERC20ETH: async (hre: HardhatRuntimeEnvironment) => {
+    const LegacyERC20ETH = await hre.ethers.getContractAt(
+      'LegacyERC20ETH',
+      predeploys.LegacyERC20ETH
+    )
+
+    const name = await LegacyERC20ETH.name()
+    if (name !== 'Ether') {
```

Comment:
General nitpick that node's built-in `assert` is perfect for this sort of stuff. `assert(name === 'Ether', 'name mismatch')`. Although for contract variables specifically you can use `assertContractVariable`.

Reply:
require vs custom errors lol